### PR TITLE
add: depracated notices added for the old methods that are depracated in version 2.8

### PR DESCRIPTION
**Pull request by guncebektas**

### CODE REVIEW

File: packages/mongo/mongo.d.ts [hunk 1]
```diff
@@ -178,17 +182,20 @@ export namespace Mongo {
     ): Promise<DispatchTransform<O['transform'], T, U> | undefined>;
     /**
      * Gets the number of documents matching the filter. For a fast count of the total documents in a collection see `estimatedDocumentCount`.
+     * @deprecated
```
Comment (StorytellerCZ) [inline packages/mongo/mongo.d.ts:185]: Incorrect, `countDocuments` is one of the replacements for `count`.

File: packages/mongo/mongo.d.ts [hunk 2]
```diff
@@ -178,17 +182,20 @@ export namespace Mongo {
...
      * @param selector The query for filtering the set of documents to count
      * @param options All options are listed in [MongoDB documentation](https://mongodb.github.io/node-mongodb-native/4.11/interfaces/CountDocumentsOptions.html). Please note that not all of them are available on the client.
      */
     countDocuments(selector?: Selector<T> | ObjectID | string, options?: NpmModuleMongodb.CountDocumentsOptions): Promise<number>;
     /**
      * Gets an estimate of the count of documents in a collection using collection metadata. For an exact count of the documents in a collection see `countDocuments`.
+     * @deprecated
```
Comment (StorytellerCZ) [inline packages/mongo/mongo.d.ts:192]: Incorrect, `estimatedDocumentCount` is one of the replacements for `count`.

**Review by StorytellerCZ — CHANGES_REQUESTED** [review]

**Review by StorytellerCZ — APPROVED** [review]

File: packages/mongo/mongo.d.ts [hunk 3]
```diff
@@ -107,6 +107,8 @@ export namespace Mongo {
       byteSize?: number,
       maxDocuments?: number
     ): Promise<void>;
+
+    /** @deprecated since 2.8 */
```
Comment (StorytellerCZ) [inline packages/mongo/mongo.d.ts:111]: After talking with @Grubba27 we would like you to add more instructions on what to use, maybe `@see createIndexAsync` or something like that. Let's use JSDocs to its full potential! 👍

**Review by StorytellerCZ — CHANGES_REQUESTED** [review]

File: packages/mongo/mongo.d.ts (hunk 3 above)
Comment (StorytellerCZ) [reply packages/mongo/mongo.d.ts:111]: I'm fine with using `@see`, I think that is cleaner.

**Review by StorytellerCZ — APPROVED** [review]

File: packages/mongo/mongo.d.ts [hunk 4]
```diff
@@ -107,6 +107,11 @@ export namespace Mongo {
       byteSize?: number,
       maxDocuments?: number
     ): Promise<void>;
+
+    /**
+     * @deprecated since 2.8. Check migration guide {@link https://guide.meteor.com/2.8-migration}
+     * @see createIndexAsync
+     */
```
Comment (Grubba27) [inline packages/mongo/mongo.d.ts:114]: I would add that is deprecated in the server only.

In the client those methods will still be available

Comment (guncebektas) [reply packages/mongo/mongo.d.ts:114]: Hmm I thought, those will be deprecated too for the client 😵‍💫 how these functions will work on methods in import folder? This topic should also be covered in the guide.

If we will continue on this pr, some of them are server only methods like createIndex. If you want, I can remove others.

**Pull request merged**